### DISCUSSION TITLE
Difference between functions with and without _tb postfix

### DISCUSSION THREAD

**Discussion by Nick-Nal** [okbob/plpgsql_check#170] [General]
Hi
Can anyone describe what's different between functions with and without _tb postfix?
Couldn't find any description in the docs about that and it's unclear to me from the code with my limited knowledge of C :)
Thanks in advance

**Comment by okbob** [comment]
ne 24. 3. 2024 v 5:55 odesílatel Nick-Nal ***@***.***> napsal:

> Hi
> Can anyone describe what's different between functions with and without
> _tb postfix?
> Couldn't find any description in the docs about that and it's unclear to
> me from the code with my limited knowledge of C :)
> Thanks in advance
>

_tb means table

 CREATE OR REPLACE FUNCTION public.buggy()
 RETURNS void
 LANGUAGE plpgsql
AS $function$
begin
  insert into nonexists values(10,20);
end;
$function$

(2024-03-24 08:12:07) postgres=# select * from
plpgsql_check_function('buggy');
┌─────────────────────────────────────────────────────────────────┐
│                     plpgsql_check_function                      │
╞═════════════════════════════════════════════════════════════════╡
│ error:42P01:3:SQL statement:relation "nonexists" does not exist │
│ Query: insert into nonexists values(10,20)                      │
│ --                 ^                                            │
└─────────────────────────────────────────────────────────────────┘
(3 rows)

(2024-03-24 08:12:27) postgres=# \x
Expanded display is on.
(2024-03-24 08:12:33) postgres=# select * from
plpgsql_check_function_tb('buggy');
┌─[ RECORD 1 ]─────────────────────────────────────┐
│ functionid │ buggy                               │
│ lineno     │ 3                                   │
│ statement  │ SQL statement                       │
│ sqlstate   │ 42P01                               │
│ message    │ relation "nonexists" does not exist │
│ detail     │ ∅                                   │
│ hint       │ ∅                                   │
│ level      │ error                               │
│ position   │ 13                                  │
│ query      │ insert into nonexists values(10,20) │
│ context    │ ∅                                   │
└────────────┴─────────────────────────────────────┘

**Reply by Nick-Nal** [reply]
Oh man, I missed the most obvious thing - return type.  Should've tried before asking :)
My bad, sorry for bothering

**Comment by okbob** [comment]
ne 24. 3. 2024 v 8:22 odesílatel Nick-Nal ***@***.***> napsal:

> Oh man, I missed the most obvious thing - return type. Should've tried
> before asking :)
> My bad, sorry for bothering
>
:-)

no problem
